include/zephyr/toolchain/iar/iccarm.h

```diff
@@ -185,6 +185,11 @@ do {                                                                    \
 #define __deprecated	__attribute__((deprecated))
 #endif
 
+#ifndef __deprecated_version
+#define __deprecated_version(version) \
+	__attribute__((deprecated("planned removal in v" #version)))
+#endif
```

Comment:
@RobinKastberg @bjorniuppsala @LoveKarlsson Could someone from IAR confirm that the `deprecated` attribute with `(msg)` is supported in IAR?